pandas/core/window/rolling.py

```diff
@@ -1110,6 +1110,35 @@ class Window(BaseWindow):
     2  2.958621
     3       NaN
     4       NaN
+
+    **on**
+
+    Rolling sum with a window length of 2 on specific columon.
+
+    >>> df = pd.DataFrame({
+    ...     'A': [pd.to_datetime('2020-01-01'),
+    ...           pd.to_datetime('2020-01-01'),
+    ...           pd.to_datetime('2020-01-02'),],
+    ...     'B': [1, 2, 3], },
+    ...     index=pd.date_range('2020', periods=3))
+
+    >>> df
+                        A  B
+    2020-01-01 2020-01-01  1
+    2020-01-02 2020-01-01  2
+    2020-01-03 2020-01-02  3
+
+    >>> df['B'].rolling('2D').sum()  # to avoid warning when sum on 'A'
+    2020-01-01    1.0
+    2020-01-02    3.0
+    2020-01-03    5.0
+    Freq: D, Name: B, dtype: float64
+
```

Comment:
Seeing as you start with

> Rolling sum with a window length of 2 on specific columon.

we can probably exclude this example. If someone tries running it locally, then can remove `on=` and see the difference, at which point it should be clear what it's done